src/diff.rs

```diff
@@ -278,6 +283,21 @@ impl<'repo> Diff<'repo> {
         Ok(buf)
     }
 
+    /// Create an patchid from a diff.
+    pub fn patchid(&self, opts: Option<&mut DiffPatchidOptions>) -> Result<Oid, Error> {
+        let mut raw = raw::git_oid {
+            id: [0; raw::GIT_OID_RAWSZ],
+        };
```

Comment:
I prefer this
```rust
let mut oid = Oid::zeros();
unsafe { try_call!(raw::git_diff_patchid(oid.raw(), ...) };
Ok(oid)
```
but all oids are constructed in raw type and wrapped before the return throughout the code.